docs/lsm/multi_version/multi_version.rst

```diff
@@ -0,0 +1,99 @@
+* Set the ``default_version`` to 0.
+* Create a ``lsm::ServiceBindingVersion`` with the information that we had on our previous binding.
+
+When unrolling using lsm::all, we use ``lsm::get_service_binding_version`` to fetch the correct entity binding version
+for each instance.
+
+.. literalinclude:: multi_version/multi_version_sources/single_version.cf
+    :linenos:
+    :language: inmanta
+    :lines: 1-49
+
+Adding or removing versions
+====================
+
+To add a new version of our service we can either create a new entity (if we want to modify the attributes of a
+previously created version) or just use the same entity but with different binding attributes
+(i.e. different lifecycle).
+
+.. literalinclude:: multi_version/multi_version_sources/multiple_versions.cf
+    :linenos:
+    :language: inmanta
+    :lines: 1-86
+
+In this example we add two new versions. Version 1 links to the same entity model but has a different lifecycle while
+Version 2 is a different entity altogether.
+When unrolling using ``lsm::all`` we can separate into groups of versions and unroll them as we wish.
+In this example, version 1 and 2 will be unrolled together into ``InterfaceIPAssignmentV2`` regardless of their original
+version, but version 0 will be unrolled separately into ``InterfaceIPAssignment``).
+
+To remove a version we can delete the corresponding ``lsm::ServiceBindingVersion`` from ``lsm::ServiceBinding.versions``
+and recompile and export the new model.
+
+NOTE: We cannot remove service entity versions that have active instances.
+
+API endpoints
+=======================
+The following API endpoints were added in order to manage versioned services:
+
+* `GET lsm/v2/service_catalog` : List all service entity versions of each defined services entity type in the service catalog
+* `POST lsm/v2/service_catalog`: Create a new service entity type in the service catalog and assign a default version to it.
+* `GET lsm/v2/service_catalog/<service_entity>/<version>/schema`: Get the json schema for a service entity version.
+* `GET lsm/v2/service_catalog/<service_entity>`: Get all versions of the service entity type from the service catalog.
+* `GET lsm/v2/service_catalog/<service_entity>/<version>`: Get one service entity version from the service catalog.
+* `PATCH lsm/v2/service_catalog/<service_entity>`: Update an existing service entity type in the service catalog or add a new one.
+* `DELETE lsm/v2/service_catalog/<service_entity>/<version>`: Delete an existing service entity version from the service catalog.
+* `GET lsm/v2/service_catalog/<service_entity>/<version>/config`: Get the config settings for a service entity version.
+* `POST lsm/v2/service_catalog/<service_entity>/<version>/config`: Set the config for a service entity version.
+* `PATCH lsm/v1/service_inventory/<service_entity>/<service_id>/update_entity_version`: Migrate a service instance from one service entity version to another.
+
```

Comment:
I miss the ones to interact with the instances.

Reply:
There are no new endpoints to interact with instances. When applicable we just have an optional `service_entity_version` argument. I'll add this information here as well.